### PR TITLE
vm/vm-under-test: Move first boot to new service

### PR DESCRIPTION
**Pull request by RamLavi**
When creating the vm-under-test container-disk image, the first-boot script running by the virt-builder command is run on a service created by virt-builder. 

This service is guaranteed to run in the final stages of the boot process [0].
This behavior may create a race with the checkup, that is waiting on the
agentConnected condition [1] being added in order to run the checkup's
executor package. 
The race is happening since the guest-agent service also runs during the final stages of the boot process.

This PR is solving this race by moving the content of the first-boot script to a new service that is guaranteed to run before the guest-agent service.

[0] https://www.libguestfs.org/virt-builder.1.html
[1] https://github.com/kiagnose/kubevirt-realtime-checkup/blob/97b5fbe47fc5df6513671db6a40e8e31b0815580/pkg/internal/checkup/checkup.go#L144

### CODE REVIEW

**Comment by RamLavi** [comment]
passes e2e test on a CNV4.15 cluster:
```

```

@orelmisan do we want to add a verbose print of the /proc/cmdline to the checkup's vm image so that we would be able to see if the cmdline ever breaks for any reason?

File: vms/vm-under-test/scripts/first-boot [hunk 1]
```diff
@@ -19,8 +19,9 @@
 
 set -e
 
-systemctl mask "$(systemctl --type swap | grep '.swap' | awk '{print $1}')"
-
+if systemctl --type swap list-units | grep -q '.swap'; then
+  systemctl mask "$(systemctl --type swap list-units | grep '.swap' | awk '{print $1}')"
+fi
```
Comment (orelmisan) [inline vms/vm-under-test/scripts/first-boot]: Could you please give a few words in the commit message about what these commands do?

File: Makefile [hunk 2]
```diff
@@ -38,7 +38,7 @@ build:
 		-v $(PWD)/_go-cache:/root/.cache/go-build:Z \
 		--workdir $(PROJECT_WORKING_DIR) \
 		$(GO_IMAGE_NAME):$(GO_IMAGE_TAG) \
-		go build -v -o ./bin/kubevirt-realtime-checkup ./cmd/
+		go build -v -buildvcs=false -o ./bin/kubevirt-realtime-checkup ./cmd/
```
Comment (orelmisan) [inline Makefile]: Leftover.

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @RamLavi 
Looks very promising.

File: vms/vm-under-test/scripts/first-boot (hunk 1 above)
Comment (RamLavi) [reply vms/vm-under-test/scripts/first-boot]: DONE

File: Makefile (hunk 2 above)
Comment (RamLavi) [reply Makefile]: fixed.

**Comment by RamLavi** [comment]
[Change](https://github.com/kiagnose/kubevirt-realtime-checkup/compare/8bae30cc9a9f2cdf9f2d5878f1218997ef27f537..ecec1c73b4de2b9812b0d714b1f0021229fb262d): Fix comments.

**Comment by orelmisan** [comment]
Does the E2E test pass?

**Review by orelmisan — APPROVED** [review]

**Comment by RamLavi** [comment]
passes e2e on CNV4.15 cluster:
```
make e2e-test
mkdir -p /home/ralavi/go/src/github.com/kiagnose/kubevirt-realtime-checkup/_go-cache
podman run --rm \
	-v /home/ralavi/go/src/github.com/kiagnose/kubevirt-realtime-checkup:/go/src/github.com/kiagnose/kubevirt-realtime-checkup:Z \
	-v /home/ralavi/go/src/github.com/kiagnose/kubevirt-realtime-checkup/_go-cache:/root/.cache/go-build:Z \
	-v /home/ralavi/.kube/sno03-cnvqe2-rdu2:/root/.kube:Z,ro \
	--workdir /go/src/github.com/kiagnose/kubevirt-realtime-checkup \
	-e KUBECONFIG=/root/.kube/kubeconfig \
	-e TEST_NAMESPACE=realtime-checkup-1 \
	-e TEST_CHECKUP_IMAGE=quay.io/ramlavi/kubevirt-realtime-checkup:devel \
	-e VM_UNDER_TEST_CONTAINER_DISK_IMAGE=quay.io/ramlavi/kubevirt-realtime-checkup-vm:latest \
	docker.io/library/golang:1.20.12 \
	go test -v ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h 
=== RUN   TestTests
Running Suite: Tests Suite - /go/src/github.com/kiagnose/kubevirt-realtime-checkup/tests
========================================================================================
Random Seed: 1704828323

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/go/src/github.com/kiagnose/kubevirt-realtime-checkup/tests/tests_suite_test.go:39
[BeforeSuite] PASSED [0.002 seconds]
------------------------------
Checkup execution should complete successfully
/go/src/github.com/kiagnose/kubevirt-realtime-checkup/tests/checkup_test.go:80
• [732.305 seconds]
------------------------------

Ran 1 of 1 Specs in 732.307 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestTests (732.31s)
PASS
ok  	github.com/kiagnose/kubevirt-realtime-checkup/tests	732.319s
```

logs show kernel is correctly updated:
```
...
2024/01/09 19:27:21 VMI under test guest kernel Args: cat /proc/cmdline
BOOT_IMAGE=(hd0,gpt2)/vmlinuz-4.18.0-348.rt7.130.el8.x86_64 root=UUID=a9332d7d-1762-41cd-a702-6b2cc556c248 ro console=tty0 rd_NO_PLYMOUTH crashkernel=auto resume=UUID=d4ac9572-e828-4c87-9b76-e59c9fa6e426 console=ttyS0,115200 skew_tick=1 tsc=reliable rcupdate.rcu_normal_after_boot=1 isolcpus=managed_irq,domain,2-3 intel_pstate=disable nosoftlockup nohz=on nohz_full=2-3 rcu_nocbs=2-3 irqaffinity=0,1
[root@realtime-vmi-under-test-jqtz4 cloud-user]# 
```

**Comment by orelmisan** [comment]
@RamLavi Please check it again after you merge the PR with `quay.io/kiagnose/kubevirt-realtime-checkup-vm:main` as the container disk image.

**Pull request merged**